homeassistant/components/sensor/ebox.py

```diff
@@ -62,25 +61,22 @@
 })
 
 
-def setup_platform(hass, config, add_devices, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
     """Set up the EBox sensor."""
     username = config.get(CONF_USERNAME)
     password = config.get(CONF_PASSWORD)
 
-    try:
-        ebox_data = EBoxData(username, password)
-        ebox_data.update()
-    except requests.exceptions.HTTPError as error:
-        _LOGGER.error("Failed login: %s", error)
-        return False
+    httpsession = hass.helpers.aiohttp_client.async_get_clientsession()
+    ebox_data = EBoxData(username, password, httpsession)
```

Comment:
Are you no longer validating that the user/pass are correct?

Reply:
I wanted to speed up the startup. If user/password are not correct, we will get an error in the logs at the next check and all sensors will be empty.
Also, Ebox website are often in maintenance, so if you restart your hass daemon when the website is unavailable, the component is disabled and you have to restart hass later.
Are you correct with this, or do you prefer to keep the user/password check ?
Thanks

Reply:
Keep the user/pass check, raise `PlatformNotReady` if it's not ready yet.

Reply:
Fixed